### PR TITLE
[GEOT-7509] Document query parameter

### PR DESCRIPTION
**Pull request by petersmythe**
[GEOT-7509](https://osgeo-org.atlassian.net/browse/GEOT-7509) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

To close https://osgeo-org.atlassian.net/browse/GEOT-7509, Bjorn and I attempted to modify the DataStore Optimisation tutorial.

We would really appreciate an informed opinion on whether this is sufficient.

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [x] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [x] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [x] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [ ] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

File: docs/user/tutorial/datastore/optimisation.rst [hunk 1]
```diff
@@ -232,7 +232,7 @@ in play.
 
 .. note:: Challenge
 
-  The ``canRetype()`` operations is easy to support, check the query and only provide values for the
+  The ``canRetype(query)`` operations are easy to support, check the query and only provide values for the
```
Comment (aaime) [inline docs/user/tutorial/datastore/optimisation.rst:235]:
```suggestion
  The ``canRetype(query)`` operation is easy to support, check the query and only provide values for the
```

**Review by aaime — COMMENTED** [review]
I'd say it looks good, but I'd also add an entry in the upgrade.rst since the old methods have been deprecated.

**Comment by bjornharrtell** [comment]
LGTM!

**Review by ianturton — APPROVED** [review]

**Comment by petersmythe** [comment]
Thank you @ianturton

**Pull request merged**